spec/java/util/stream/Stream.main.lsl

```diff
@@ -0,0 +1,1049 @@
+
+
+    fun *.min (@target self: Stream, comparator: Comparator): Optional
+    {
+        if (comparator == null)
+            _throwNPE();
+
+        if (this.length == 0)
+        {
+            result = action DEBUG_DO("Optional.empty()");
+        }
+        else
+        {
+            var min: Object = this.storage[0];
+
+            var i: int = 0;
+            action LOOP_FOR(
+                i, 1, this.length, +1,
+                _find_min_loop(i, comparator, min)
+            );
+
+            result = action DEBUG_DO("Optional.of(min)");
+        }
+    }
+
+
+    @Phantom proc _find_min_loop (i: int, comparator: Comparator, min: int): void
+    {
+        if (action CALL(comparator, [min, this.storage[i]]) > 0)
+            min = this.storage[i];
+    }
+
+
+    fun *.max (@target self: Stream, comparator: Comparator): Optional
+    {
+        if (comparator == null)
+            _throwNPE();
+
+        if (this.length == 0)
+        {
+            result = action DEBUG_DO("Optional.empty()");
+        }
+        else
+        {
+            var max: Object = this.storage[0];
+
+            var i: int = 0;
+            action LOOP_FOR(
+                i, 1, this.length, +1,
+                _find_max_loop(i, comparator, max)
+            );
+
+            result = action DEBUG_DO("Optional.of(max)");
+        }
+    }
+
+
+    @Phantom proc _find_max_loop (i: int, comparator: Comparator, max: int): void
+    {
+        if (action CALL(comparator, [max, this.storage[i]]) < 0)
+            max = this.storage[i];
+    }
+
+
+    fun *.count (@target self: Stream): long
+    {
+        result = this.length;
+    }
+
+
+    fun *.anyMatch (@target self: Stream, predicate: Predicate): boolean
+    {
+        if (predicate == null)
+            _throwNPE();
+
+        result = false;
+
+        var i: int = 0;
+        action LOOP_FOR(
+            i, 0, this.length, +1,
+            _anyMatch_loop(i, predicate, result)
+        );
+    }
+
+
+    @Phantom proc _anyMatch_loop (i: int, predicate: Predicate, result: boolean): void
+    {
+        if (action CALL(predicate, [this.storage[i]]))
+        {
+            result = true;
+            action LOOP_BREAK();
+        }
+    }
+
+
+    fun *.allMatch (@target self: Stream, predicate: Predicate): boolean
+    {
+        if (predicate == null)
+            _throwNPE();
+
+        result = true;
+
+        var i: int = 0;
+        action LOOP_FOR(
+            i, 0, this.length, +1,
+            _allMatch_loop(i, predicate, result)
+        );
+    }
+
+
+    @Phantom proc _allMatch_loop (i: int, predicate: Predicate, result: boolean): void
+    {
+        if (!action CALL(predicate, [this.storage[i]]))
+        {
+            result = false;
+            action LOOP_BREAK();
+        }
+    }
+
+
+    fun *.noneMatch (@target self: Stream, predicate: Predicate): boolean
+    {
+        if (predicate == null)
+            _throwNPE();
+
+        result = true;
+
+        var i: int = 0;
```

Comment:
What if there no elements in this stream?

Reply:
Fixed